daemon/src/model/cfd.rs

```diff
@@ -2373,6 +2390,219 @@ mod tests {
+
+        let cfd = cfd.dummy_start_collab_settlement();
+
+        let err = cfd.start_rollover().unwrap_err();
+        let no_rollover_reason = err.downcast::<NoRolloverReason>().unwrap();
+        assert_eq!(
+            no_rollover_reason,
+            NoRolloverReason::InCollaborativeSettlement
+        );
+    }
+
+    #[test]
+    fn given_ongoing_rollover_then_can_start_collaborative_settlement() {
+        let quantity = Usd::new(dec!(10));
+        let leverage = Leverage::new(2).unwrap();
+        let opening_price = Price::new(dec!(10000)).unwrap();
+        let order_id = OrderId::default();
+
+        let taker_keys = crate::keypair::new(&mut rand::thread_rng());
+        let maker_keys = crate::keypair::new(&mut rand::thread_rng());
+
+        let taker_long = Cfd::taker_long()
+            .with_id(order_id)
+            .with_quantity(quantity)
+            .with_opening_price(opening_price)
+            .with_leverage(leverage)
+            .dummy_open(dummy_event_id());
+
+        let (taker_long, proposal, taker_sig, _) = taker_long
+            .with_lock(taker_keys, maker_keys)
+            .dummy_start_rollover()
+            .dummy_collab_settlement_taker(opening_price);
+
+        let maker_short = Cfd::maker_short()
+            .with_id(order_id)
+            .with_quantity(quantity)
+            .with_opening_price(opening_price)
+            .with_leverage(leverage)
+            .dummy_open(dummy_event_id());
+
+        let (maker_short, _) = maker_short
+            .with_lock(taker_keys, maker_keys)
+            .dummy_start_rollover()
+            .dummy_collab_settlement_maker(proposal, taker_sig);
+
+        assert!(
+            taker_long.collaborative_settlement_spend_tx.is_some(),
+            "No settlement tx even though the settlement passed"
+        );
+        assert!(
+            maker_short.collaborative_settlement_spend_tx.is_some(),
+            "No settlement tx even though the settlement passed"
+        );
+    }
+
+    #[test]
+    fn given_collab_settlement_then_cannot_force_close() {
+        let quantity = Usd::new(dec!(10));
+        let leverage = Leverage::new(2).unwrap();
+        let opening_price = Price::new(dec!(10000)).unwrap();
+
+        let taker_keys = crate::keypair::new(&mut rand::thread_rng());
+        let maker_keys = crate::keypair::new(&mut rand::thread_rng());
+
+        let cfd = Cfd::taker_long()
+            .with_quantity(quantity)
+            .with_opening_price(opening_price)
+            .with_leverage(leverage)
+            .dummy_open(dummy_event_id());
+
+        let (cfd, _, _, _) = cfd
+            .with_lock(taker_keys, maker_keys)
+            .dummy_collab_settlement_taker(opening_price);
+
+        // TODO: Potentially change the API so we can assert the actual error
+        assert!(
+            cfd.manual_commit_to_blockchain().is_err(),
+            "Manual commit to blockchain did not error"
+        );
```

Comment:
Why not `unwrap_err` and assert on the error, i.e. call `to_string` on it and assert on the reason?